crates/host/src/wasmbus/mod.rs

```diff
@@ -199,53 +139,33 @@ impl Queue {
         host_key: &KeyPair,
     ) -> anyhow::Result<Self> {
         let host_id = host_key.public_key();
-        let (
-            registries,
-            pings,
-            links,
-            queries,
-            auction,
-            commands,
-            inventory,
-            labels,
-            config,
-            config_get,
-        ) = try_join!(
-            nats.subscribe(format!("{topic_prefix}.{lattice}.registries.put",)),
-            nats.subscribe(format!("{topic_prefix}.{lattice}.ping.hosts",)),
-            nats.queue_subscribe(
-                format!("{topic_prefix}.{lattice}.linkdefs.*"),
-                format!("{topic_prefix}.{lattice}.linkdefs",)
-            ),
-            nats.queue_subscribe(
-                format!("{topic_prefix}.{lattice}.get.*"),
-                format!("{topic_prefix}.{lattice}.get")
-            ),
-            nats.subscribe(format!("{topic_prefix}.{lattice}.auction.>",)),
-            nats.subscribe(format!("{topic_prefix}.{lattice}.cmd.{host_id}.*",)),
-            nats.subscribe(format!("{topic_prefix}.{lattice}.get.{host_id}.inv",)),
-            nats.subscribe(format!("{topic_prefix}.{lattice}.labels.{host_id}.*",)),
-            nats.queue_subscribe(
+        let streams = futures::future::join_all([
+            Either::Left(nats.subscribe(format!("{topic_prefix}.{lattice}.registry.put",))),
+            Either::Left(nats.subscribe(format!("{topic_prefix}.{lattice}.host.ping",))),
+            Either::Left(nats.subscribe(format!("{topic_prefix}.{lattice}.*.auction",))),
+            Either::Right(nats.queue_subscribe(
+                format!("{topic_prefix}.{lattice}.link.*"),
+                format!("{topic_prefix}.{lattice}.link",),
+            )),
```

Comment:
Ooh are the left and rights here a clever way to get around type differences with queue and regular subscriptions?

Reply:
Yes! I discovered the `Either` type in futures and wondered why I hadn't learned about it sooner